src/App.tsx

```diff
@@ -1,60 +1,115 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { ArrayOfTodos, Todos } from './types/Todos';
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId)
+    || null;
+}
+
+export const todosWithUser: ArrayOfTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<ArrayOfTodos>(todosWithUser);
+  const [title, setTitle] = useState('');
+  const [userId, setUserID] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const handleUserIdChange = (event:
+  React.ChangeEvent<HTMLSelectElement>) => {
+    setUserID(+event.target.value);
+    setUserError(false);
+  };
+
+  const addTodos = (newTodo: Todos) => {
```

Comment:
Only one todo is added by this function
```suggestion
  const addTodo = (newTodo: Todos) => {
```